Argo/Types/Decoded.swift

```diff
@@ -102,8 +111,8 @@ public func <^> <A, B>(f: A -> B, a: Decoded<A>) -> Decoded<B> {
 
 /**
   apply a `Decoded` function to a `Decoded` value
-  - If the function is a failing case (`.TypeMismatch`, `.MissingKey`), this will return the function's value
-  - If the function is a `.Success` case and the value is a failure case, this will return the value
+  - If the function is `.Failure`, this will return the function's failure info
+  - If the function is a `.Success` case and the value is `Failure`, this will return the value's failure info
```

Comment:
I know this was like this before these changes but I think we lost some consistency with this
maybe have it more like this: `If the function is `.Success` and the ...`